### PR TITLE
fix: stop containers more thoroughly

### PR DESCRIPTION
**Pull request by smira**
Don't skip pods which are not ready, try still to stop containers inside not ready pod sandboxes.

Re-enable the test with Canal CNI (upstream Calico got fixed).

Signed-off-by: Andrey Smirnov <andrey.smirnov@talos-systems.com>

### CODE REVIEW

File: internal/pkg/cri/pods.go [hunk 1]
```diff
@@ -182,18 +178,22 @@ func stopAndRemove(ctx context.Context, stopAction StopAction, client *Client, p
...
 
 		g.Go(func() error {
-			log.Printf("%s container %s/%s:%s", action, pod.Metadata.Namespace, pod.Metadata.Name, container.Metadata.Name)
-
 			// TODO(andrewrynhard): Can we set the timeout dynamically?
-			if criErr := client.StopContainer(ctx, container.Id, 30); criErr != nil {
-				if grpcstatus.Code(criErr) == codes.NotFound {
-					return nil
-				}
+			if container.State == runtimeapi.ContainerState_CONTAINER_RUNNING || container.State == runtimeapi.ContainerState_CONTAINER_UNKNOWN {
```
Comment (frezbo) [inline internal/pkg/cri/pods.go:182]: do we have any potential other states in between as reported by cri, maybe a container is still restarting/not healthy?

File: internal/pkg/cri/pods.go [hunk 2]
```diff
@@ -213,14 +213,16 @@ func stopAndRemove(ctx context.Context, stopAction StopAction, client *Client, p
 		return err
 	}
 
-	if err = client.StopPodSandbox(ctx, pod.Id); err != nil {
-		if grpcstatus.Code(err) == codes.NotFound {
-			return nil
-		}
+	if pod.State == runtimeapi.PodSandboxState_SANDBOX_READY {
```
Comment (frezbo) [inline internal/pkg/cri/pods.go:216]: what if a pod failed to start due to cni error, is it okay to go ahead?

Comment (smira) [reply internal/pkg/cri/pods.go:216]: we just don't try to stop a pod if it's not ready, as if it's not ready, it got stopped already

File: internal/pkg/cri/pods.go (hunk 1 above)
Comment (smira) [reply internal/pkg/cri/pods.go:182]: these are two different states, there's a sandbox (pod) state and a container state. 

we are in fact interested in stopping containers, as they hold the mounts, etc. 

we stop the pods as well, but that mostly means tearing down CNI stuff

**Review by frezbo — APPROVED** [review]

**Pull request merged**